include/blmc_robots/n_joint_blmc_robot_driver.hpp

```diff
@@ -80,6 +78,199 @@ class NJointBlmcRobotDriver
+            .move_timeout = 0,
+        };
+
+        //! \brief D-gain to dampen velocity.  Set to zero to disable damping.
+        // set some rather high damping by default
+        Vector safety_kd = Vector::Constant(0.1);
+
+        //! @brief Default control gains for the position PD controller.
+        struct
+        {
+            Vector kp = Vector::Zero();
+            Vector kd = Vector::Zero();
+        } position_control_gains;
+
+        //! \brief Offset between home position and zero.
+        Vector home_offset_rad = Vector::Zero();
+
+        /**
+         * @brief Initial position to which the robot moves after
+         *        initialization.
+         */
+        Vector initial_position_rad = Vector::Zero();
+
+        /**
+         * @brief Print the given configuration in a human-readable way.
+         */
+        void print() const
+        {
+            std::cout << "Configuration:\n"
+                      << "\t max_current_A: " << max_current_A << "\n"
+                      << "\t has_endstop: " << has_endstop << "\n"
+                      << "\t calibration: "
+                      << "\n"
+                      << "\t\t torque_ratio: " << calibration.torque_ratio
+                      << "\n"
+                      << "\t\t position_tolerance_rad: "
+                      << calibration.position_tolerance_rad << "\n"
+                      << "\t\t move_timeout: " << calibration.move_timeout
+                      << "\n"
+                      << "\t safety_kd: " << safety_kd.transpose() << "\n"
+                      << "\t position_control_gains: "
+                      << "\n"
+                      << "\t\t kp: " << position_control_gains.kp.transpose()
+                      << "\n"
+                      << "\t\t kd: " << position_control_gains.kd.transpose()
+                      << "\n"
+                      << "\t home_offset_rad: " << home_offset_rad.transpose()
+                      << "\n"
+                      << "\t initial_position_rad: "
+                      << initial_position_rad.transpose() << "\n"
+                      << std::endl;
+        }
+
+        /**
+         * @brief Load driver configuration from file or using default values.
+         *
+         * Load the configuration from the specified YAML file.  The file is
+         * expected to have the same structure/key naming as the Config struct.
+         * For entries that are not specified in the file the value from
+         * default_config is used.
+         *
+         * @param config_file_name  Path/name of the configuration YAML file.
+         * @param default_config  Default values that are used for fields that
+         *     are not specified in the configuration file.
+         *
+         * @return Configuration
+         */
+        static Config load_config(const std::string &config_file_name,
+                                  const Config &default_config)
+        {
```

Comment:
this is super nice that we can load configs from file! what i am wondering: maybe we should not even have a default config, maybe everything should be loaded from the file. I think this might be less error-prone, because this way all the information is in one place, whereas otherwise people might forget to put something in the config file, or have some typo and the code still runs through.

Reply:
I am unsure about this. It is right that it is a bit error-prone at the moment (e.g. when having a typo in a field in the config file, it will silently be ignored and the default is used instead. On the other hand, putting everything into the config file would result in lots of redundancy once we have multiple platforms where most of the parameters will be identical but a few will differ, so we need a separate file for each.

Reply:
Result of offline discussion: I will remove the default values here and throw errors for missing values.
To handle the maintenance issue, we can later split it into separate files, one for the parameters that are expected to differ between platforms (e.g. homing offset) and one for the ones that will be the same everywhere.